games/static/games/proto.js

```diff
@@ -99,16 +127,13 @@ class Defender {
 }
 
 class Rocket {
-    constructor(imageCssClass, indexOnBoard, board) {
+    constructor(imageCssClass, index, board) {
         this.imageCssClass = imageCssClass
-        this.indexOnBoard = indexOnBoard
+        this.index = index
         this.board = board
     }
-    fly() {
-        let x = this.board.squares[this.indexOnBoard].pickCharacterUp()
-        this.indexOnBoard = this.upTheBoard()
-        this.board.squares[this.indexOnBoard].setCharacter(x)
-
+    isAlien() {
```

Comment:
can you explain ?

Reply:
yes, sometimes the rockets would get on their interval an alien and then the alien would fly up the board instead of the rocket so i check and if the square in the interval is an alien i set it back to rocket

Reply:
but if i checked and it was not an alien i got an error in the console so i did this method on alien